src/wait.ts

```diff
@@ -1,43 +1,61 @@
-import { Run } from "./github";
+import { Run, OctokitGitHub, GitHub } from "./github";
+import { Input, parseInput } from "./input";
 
 export interface Wait {
   wait(secondsSoFar?: number): Promise<number>;
 }
 
 export class Waiter implements Wait {
   private readonly info: (msg: string) => void;
-  private readonly getRun: () => Promise<Run>;
-  private readonly pollIntervalSeconds: number;
-  private readonly continueAfterSeconds: number | undefined;
+  private input: Input;
+  private githubClient: GitHub;
+  private workflowId: any;
+
   constructor(
-    getRun: () => Promise<Run>,
-    pollIntervalSeconds: number,
-    continueAfterSeconds: number | undefined,
+    workflowId: any,
+    githubClient: GitHub,
+    input: Input,
     info: (msg: string) => void
   ) {
-    this.getRun = getRun;
-    this.pollIntervalSeconds = pollIntervalSeconds;
-    this.continueAfterSeconds = continueAfterSeconds;
+    this.workflowId = workflowId;
+    this.input = input;
+    this.githubClient = githubClient;
     this.info = info;
   }
 
   wait = async (secondsSoFar?: number) => {
+    const runs = await this.githubClient.runs(
```

Comment:
Can we move this API call below the check for exceeded seconds. If that condition is met this will have been a wasted api call

Reply:
Yep. Done.